python/whatsapp_with_vonage/src/main.py

```diff
@@ -0,0 +1,65 @@
+from .utils import get_static_file, throw_if_missing
+import os
+from jwt import decode
+from requests import post
+
+
+def main(context):
+    throw_if_missing(
+        dict(os.environ),
+        [
+            "VONAGE_API_KEY",
+            "VONAGE_API_SECRET",
+            "VONAGE_API_SIGNATURE_SECRET",
+            "VONAGE_WHATSAPP_NUMBER",
+        ],
+    )
+
+    if context.req.method == "GET":
+        return context.res.send(
+            get_static_file("index.html"),
+            200,
+            {"content-type": "text/html"},
+        )
+
+    body = context.req.body
+    headers = context.req.headers
+    token = (headers["authorization"] or "").split(" ")[1]
+    try:
+        decode(token, os.environ["VONAGE_API_SIGNATURE_SECRET"], ["HS256"])
+    except Exception as e:
+        return context.res.json({"ok": False, "error": "Invalid token!"}, 403)
```

Comment:
I would recommend error message to be `Payload hash mismatch`, just like in Node version.